cmd/run_helpers.go

```diff
@@ -0,0 +1,340 @@
+package cmd
+
+import (
+	"os"
+	"strings"
+	"log"
+	"errors"
+	"fmt"
+	"time"
+	"sort"
+	"regexp"
+
+	"github.com/spf13/viper"
+	"github.com/gobwas/glob"
+)
+
+func isSkipCommmand(hooksGroup, executableName string) bool {
+	key := strings.Join([]string{hooksGroup, commandsConfigKey, executableName, skipConfigKey}, ".")
+	value := viper.Get(key)
+
+	switch typedValue := value.(type) {
+	case bool:
+		/*
+		pre-push:
+			commands:
+				packages-audit:
+					skip: true
+		*/
+		return typedValue
+	case string:
+		/*
+		pre-push:
+			commands:
+				packages-audit:
+					skip: merge
+		*/
+		return isSkippedGitState(typedValue)
+	case []interface{}:
+		/*
+		pre-push:
+			commands:
+				packages-audit:
+					skip:
+						- merge
+						- rebase
+		*/
+		for _, gitState := range typedValue {
+			if isSkippedGitState(gitState.(string)) {
+				return true
+			}
+		}
+	}
+
+	return false
+}
+
+func isSkippedGitState(state string) bool {
+	return state == "merge" && isMergeInProgress() || state == "rebase" && isRebaseInProgress()
+}
+
+func isMergeInProgress() bool {
+	if _, err := os.Stat(".git/MERGE_HEAD"); os.IsNotExist(err) {
+		return false
+	}
+
+	return true
+}
+
+func isRebaseInProgress() bool {
+	if _, mergeErr := os.Stat(".git/rebase-merge"); os.IsNotExist(mergeErr) {
```

Comment:
Here is another way of checking rebasing state (not tested personally, though):
https://github.com/istio/old_mixer_repo/pull/262/files#diff-7ee9e529af99bc938983416b1810255ecd127a0f928d5c7c9d9c9d8359593d3fR28

Reply:
Detached HEAD state may occur in many ways. User can manually go to it executing `git checkout somecommitsha`, so no, we shouldn't rely on it.

Reply:
@Envek what do you think of my approach?

Reply:
I think that it is totally right. Not sure about details: should we check for these `rebase-merge` and `rebase-apply` directories or for `REBASE_HEAD` file.